app/controllers/api/v1/packages_controller.rb

```diff
@@ -0,0 +1,70 @@
+class Api::V1::PackagesController < ApplicationController
+  before_action :set_package, only: %i[show edit update destroy]
+
+  # GET /packages or /packages.json
+  def index
+    @packages = Package.all
+    render json: @packages
+  end
+
+  # GET /packages/1 or /packages/1.json
+  def show; end
+
+  # GET /packages/new
+  def new
+    @package = Package.new
+  end
+
+  # GET /packages/1/edit
+  def edit; end
+
+  # POST /packages or /packages.json
+  def create
+    @package = Package.new(package_params)
+
+    respond_to do |format|
+      if @package.save
+        format.html { redirect_to package_url(@package), notice: 'Package was successfully created.' }
+        format.json { render :show, status: :created, location: @package }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @package.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /packages/1 or /packages/1.json
+  def update
+    respond_to do |format|
+      if @package.update(package_params)
+        format.html { redirect_to package_url(@package), notice: 'Package was successfully updated.' }
+        format.json { render :show, status: :ok, location: @package }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @package.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /packages/1 or /packages/1.json
+  def destroy
+    @package.destroy
```

Comment:
Missing error handling here, what happens if it does not work the way you expect it to?